### PR TITLE
Some correction around PseudoParameterBinder

### PR DESCRIPTION
**Pull request by kwkam**
## PR Summary

_*Automation/engine/CommandCompletion/CompletionCompleters:_
Add a missing null checking

_*Automation/engine/CommandDiscovery:_
LookupCommandInfo would return null, also checks for it

Command like `& "$XXX_HOME\` should now be tab completable

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] `NA` User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] `NA` Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] `NA` [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] `NA` [Add `[feature]` if the change is significant or affectes feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by iSazonov** [comment]
@kwkam  Thanks for your contribution! Please follow the requirements of PR Checklist.

File: src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs [hunk 1]
```diff
@@ -1075,16 +1075,16 @@ internal PseudoBindingInfo DoPseudoParameterBinding(CommandAst command, Type pip
         /// <param name="executionContext">ExecutionContext</param>
         private void SetTemporaryDefaultHost(ExecutionContext executionContext)
         {
-            if (executionContext.EngineHostInterface.IsHostRefSet)
+            if (_restoreHost == null && executionContext.EngineHostInterface.IsHostRefSet)
```
Comment (daxian-dbw) [inline src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs]: I don't see the problem with the current code, can you please explain why the change in this method is needed?

**Review by daxian-dbw — CHANGES_REQUESTED** [review]
> Also, command like:
& "$XXX_HOME\
should now be tab completable

Your changes seem to fix some user-faced issues, so please add tests too.

File: src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs (hunk 1 above)
Comment (iSazonov) [reply src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs]: Please revert.

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs]: https://github.com/kwkam/PowerShell/blob/8f98f36f533d4f94679828b5f03517d72e140a2d/src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs#L968-L989

Comment (daxian-dbw) [reply src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs]: By looking at the code where this method is used, `RestoreHost` is called in the `finally` block. And they are manipulating the `executioncontext` from the thread local storage.

It looks to me the variable won't be overwritten as long as you are not passing the same `PseudoParameterBinder` instance to multiple pipeline threads and calling methods on them at the same time.

A `PseudoParameterBinder` instance is not supposed to be shared between pipeline threads, so I don't think we need to change it here. Please revert this change.

https://github.com/kwkam/PowerShell/blob/8f98f36f533d4f94679828b5f03517d72e140a2d/src/System.Management.Automation/engine/CommandCompletion/PseudoParameterBinder.cs#L968-L989

**Comment by kwkam** [comment]
@daxian-dbw @iSazonov Yes reverted.

**Review by daxian-dbw — APPROVED** [review]

**Comment by iSazonov** [comment]
LGTM.
